src/main/com/fulcrologic/fulcro/routing/dynamic_routing.cljc

```diff
@@ -934,15 +935,21 @@
                                                                               :current-state        ~'current-state}]
                                                        ~@body)))))
            options                (merge
-                                    `{:componentDidMount (fn [this#] (validate-route-targets this#))}
+                                    `{:componentDidMount (fn [this#] (validate-route-targets this#))
```

Comment:
so if use hooks is true, you just leave this hook here, and are happy because it is a no-op? It's a little dangerous in that what if we add error checking to defsc around hooks to warn you when you do something dumb? Maybe a cond-> or something?

Reply:
Something like [this?](https://github.com/fulcrologic/fulcro/compare/8c01463a88fb867bfe18229629d1cd3c6f3c860e..c8dbb7851f5321bce91c15f9f00bf3b96b75dc92#diff-dbc631589940e7571a88fa512a517d49cc6e731bcaaa85c04b95de1c4d7df07bR954)

Reply:
yes